drivers/digital-io/max149x6/max14906.c

```diff
@@ -199,6 +199,40 @@ int max14906_init(struct max149x6_desc **desc,
 			goto gpio_err;
 	}
 
+	ret = no_os_gpio_get_optional(&descriptor->fault_gpio,
+				      param->fault_gpio_param);
+	if (ret)
+		goto spi_err;
+
+	if (descriptor->fault_gpio) {
+		ret = no_os_gpio_direction_input(descriptor->fault_gpio);
+		if (ret)
+			goto gpio_err;
+	}
+
+	ret = no_os_gpio_get_optional(&descriptor->ready_gpio,
+				      param->synch_gpio_param);
+	if (ret)
+		goto spi_err;
+
+	if (descriptor->ready_gpio) {
+		ret = no_os_gpio_direction_input(descriptor->ready_gpio);
+		if (ret)
+			goto gpio_err;
+	}
+
+	ret = no_os_gpio_get_optional(&descriptor->synch_gpio,
+				      param->synch_gpio_param);
+	if (ret)
+		goto spi_err;
+
+	if (descriptor->synch_gpio) {
+		ret = no_os_gpio_set_value(descriptor->synch_gpio,
```

Comment:
You have to set this gpio as output first

Reply:
Fixing it.